zerver/lib/outgoing_webhook.py

```diff
@@ -32,8 +32,8 @@ def build_bot_request(self, event: Dict[str, Any]) -> Optional[Any]:
         # outgoing webhook to indicate whether it wants the raw
         # Markdown or the rendered HTML, we leave both the content and
         # rendered_content fields in the message payload.
-        MessageDict.finalize_payload(event['message'], False, False,
-                                     keep_rendered_content=True)
+        event['message'] = MessageDict.finalize_payload(event['message'], False, False,
+                                                        keep_rendered_content=True)
```

Comment:
Is it correct to store as `event['message']` here, or do we want to use a different local variable?  

I guess I don't understand under what circumstance we end up calling `finalize_payload` twice in the previous code; can you explain that detail?